### PR TITLE
Analysis progress reporting.

### LINKED ISSUE #370: Add some progress report to analyzer info messages
Big projects can take hours to finish, and it would be nice to know at least approximately how much stuff there is left to analyse.

As a first step I was thinking about a simple (ninja style) indicator on how many steps are left from the analysis process.

Something like:

```
[...]
[INFO] - [34/705] clangsa analyzed A.cpp successfully.
[INFO] - [35/705] clangsa analyzed B.cpp successfully.
[INFO] - [36/705] clangsa analyzed C.cpp successfully.
[INFO] - [37/705] clangsa analyzed D.cpp successfully.
[INFO] - [38/705] clangsa analyzed E.cpp successfully.
[INFO] - [39/705] clangsa analyzed F.cpp successfully.
[...]
```

wdyt?

### PR DESCRIPTION
**Pull request by Xazax-hun**
Fixes  #370.

Some notes to the changes:
- Shared memory values can not be passed as function arguments, that is the reason for the globals.
    - They could be moved into an empty imported module
- Result handler no longer prints successful analysis message. It was asymmetric. Now successful and unsuccessful results are printed from the same location.
- Result handler only invoked to process results, when the analysis succeeded. Updated the code to reflect this.
- Invoke the process pool implementation from the quickcheck command. This way multiprocess checking with quickcheck can be implemented in the future. (We might need to lock the standard output for concurrent writes.)

### CODE REVIEW

File: codechecker_lib/analysis_manager.py [hunk 1]
```diff
@@ -155,7 +174,7 @@ def signal_handler(*arg, **kwarg):
     signal.signal(signal.SIGINT, signal_handler)
 
     # Remove characters which could cause directory creation problems.
-    no_spec_char_name = re.sub(r'[^\w\-_\. ]', '_', args.name)
+    no_spec_char_name = re.sub(r'[^\w\-_. ]', '_', args.name)
```
Comment (gyorb) [inline codechecker_lib/analysis_manager.py]: Why is this change needed? Just a temporary directory is created. The ```.``` char is replaced with ```_```.

Comment (Xazax-hun) [inline codechecker_lib/analysis_manager.py]: It is just a cleanup, in that context no escaping of `.` required.

Comment (Xazax-hun) [inline codechecker_lib/analysis_manager.py]: This is not a functional change just a cleanup. Dot was *not* replaced befor this modification and it is replaced afterwards.

Comment (gyorb) [inline codechecker_lib/analysis_manager.py]: Ok, my bad I misinterpreted the regex.
